networking_aci/plugins/ml2/drivers/mech_aci/allocations_manager.py

```diff
@@ -42,7 +44,29 @@
 class AllocationsManager(object):
     def __init__(self, db):
         self.db = db
+        self._sync_db_guarded()
 
+    def initialize(self):
+        self._sync_db_guarded()
+        LOG.info(_LI("AllocationsManager initialization complete"))
+
+    def _sync_db_guarded(self):
+        if not CONF.ml2_aci.sync_allocations_done_file_path:
+            # sync db unconditionally
+            self._sync_db()
+            return
+
+        with lockutils.external_lock("sync-allocations-guard", lock_file_prefix="networking-aci"):
+            try:
+                synced_by = open(CONF.ml2_aci.sync_allocations_done_file_path, "r").read()
+                LOG.info("Not syncing in this thread, already synced by PID %s", synced_by)
+            except FileNotFoundError:
+                self._sync_db()
+                with open(CONF.ml2_aci.sync_allocations_done_file_path, "w") as f:
+                    f.write(str(os.getpid()))
+                LOG.info("Successfully synced db")
```

Comment:
Hmm you're thinking about tracking the time for the whole call? We could do that. Currently you could look at the two timestamps and subtract the time, but a direct output for it might be nice.